src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,80 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void serviceInitialization() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void userStorageInitialization() {
+        user = new User(1L,"valid_login", "123456", 40);
+        Storage.people.addAll(List.of(new User(2L,"LarsUrlich","myolddrums",60),
+                new User(2L, "JamesHetfield","mastermaster",61),
+                new User(3L,"KirkHammet","Imtoofastforyou",60),
+                new User(4L,"JasonNewsted","imnothere777",99)));
+    }
+
+    @Test
+    void nullLoginIfExceptionThrow_Ok() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));

```

Comment:
The test method name `nullLoginIfExceptionThrow_Ok` does not follow the convention `<methodUnderTest>_<state>_<expectedBehavior>`. A better name would be `register_nullLogin_notOk` to reflect the method under test (`register`), the state being tested (`nullLogin`), and the expected behavior (`notOk`).

Reply:
ok, done